### PR TITLE
fix: fix contact not updating when an ID verif is caneled

### LINKED ISSUE #13412: Rejected identity verification request is displayed as pending on user's profile
# Bug Report

## Description

When a identity verification request is rejected by the recipient, the contact request is still displayed as pending when looking at their profile. In the notification tab it's displayed as rejected. This makes it impossible to send a new verification request to that same contact, because it makes it impossible to cancel a request in that case. Cancelling a request that has not been rejected works as intended.

## Steps to reproduce

1. Send a contact request and accept
2. Send a ID request to the contact
3. Recipient of ID request refuses
4. ID request is shown as refused in activity centre

#### Expected behavior

- ID request is not shown as pending in profile menu
- Cancel button cancels ID request
- ID request shouldn't be displayed as pending when it has been rejected?

#### Actual behavior

- ID request is shown as pending in profile menu
- Cancel button does not cancel the ID request

### Additional Information

- Status desktop version: Can be found here: https://ci.status.im/job/status-desktop/job/systems/job/macos/job/aarch64/job/package/284/
- Operating System: macOS

https://github.com/status-im/status-desktop/assets/155960867/fdee120f-be2a-451c-be45-adec5c4ea881

Console log of cancel button not working:
"ERR 2024-02-02 16:32:15.777+00:00 rpc response error                         topics="rpc" tid=27204477 file=core.nim:36 err="\nstatus-go error [methodName:wakuext_cancelVerificationRequest, code:-32000, message:can cancel only pending verification request ]\n"
ERR 2024-02-02 16:32:15.777+00:00 error doing rpc request                    topics="rpc" tid=27204477 file=core.nim:40 methodName=wakuext_cancelVerificationRequest exception="\nstatus-go error [methodName:wakuext_cancelVerificationRequest, code:-32000, message:can cancel only pending verification request ]\n"
ERR 2024-02-02 16:32:15.777+00:00 Error canceling verification request       topics="contacts-service" tid=27204477 file=service.nim:754 msg="\nstatus-go error [methodName:wakuext_cancelVerificationRequest, code:-32000, message:can cancel only pending verification request ]\n""

### PR DESCRIPTION
**Pull request by jrainville**
Fixes https://github.com/status-im/status-desktop/issues/13412

### CODE REVIEW

File: protocol/messenger_contact_verification.go [hunk 1]
```diff
@@ -962,6 +962,24 @@ func (m *Messenger) HandleDeclineContactVerification(state *ReceivedMessageState
 		return nil // Do nothing, We have already cancelled the verification request
 	}
 
+	contact.VerificationStatus = VerificationStatusUNVERIFIED
```
Comment (MishkaRogachev) [inline protocol/messenger_contact_verification.go:969]: Not sure about this. Identity verification flows are not bi-directional, and we already mark contacts as unverified in  [CancelVerificationRequest](https://github.com/status-im/status-go/blob/develop/protocol/messenger_contact_verification.go#L182). But maybe requirements have changed and i'm not aware of it

**Review by MishkaRogachev — COMMENTED** [review]
One question

**Review by mprakhov — COMMENTED** [review]
Looks ok for me, but maybe we can cover this case with tests if this changes did not impact none of your tests

File: protocol/messenger_contact_verification.go (hunk 1 above)
Comment (jrainville) [reply protocol/messenger_contact_verification.go:969]: You're right, I forgot to think about the flows being bi-directional.

Good thing for me though, this is still correct as far as I understand.

We use `receivedVerifications` for the verifications that comes from other users. And we use the `contact.VerificationStatus` for the ones we send.

So in this case, we sent the verification to user B, so we marked the `VerificationStatus` as `VerificationStatusVERIFYING`. 
Then user B canceled, so the status goes back to the default one of `VerificationStatusUNVERIFIED`.

This doesn't affect the verification status from recived verifications. If user B sent us a verification and we sent back a response and they approved it, we would be `VerificationStatusVERIFIED` for them.

**Review by MishkaRogachev — APPROVED** [review]
LGTM!

**Comment by jrainville** [comment]
@mprakhov done. I improved the test to check for the contact's verification status. I added the check on the other tests too

**Review by mprakhov — APPROVED** [review]
